### PR TITLE
created detect_iodine function

### PR DESCRIPTION
**Pull request by nicholustintzaw**
finished developing `detect_iodine` function with the test package. All check works were done with no errors or warnings.

### CODE REVIEW

File: R/data.R [hunk 1]
```diff
@@ -30,3 +30,71 @@
...
+#' head(iodine_sample)
+#'
+#
+################################################################################
+"iodine_sample"
+
+################################################################################
+#
+#'
+#' Urinary iodine (μg/L) values for women and children
```
Comment (ernestguevarra) [inline R/data.R:70]: update title of this dataset to reflect serum ferritin

File: R/detect_iodine.R [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+#'    | School-age-children (6 years or odler) |	"general" |
+#'    | Pregnant Women |	"pregnant" |
+#'    | Lactating Mothers and Children under 2 years old	| "lactating" |
+#'
+#'    Please note that the group parameter only accepts a single value of the
+#'    specific group's name, not the combination of more than one groups' name
+#'    in vector form. (for example, group == "general")
+#'
+#'
+#' @return iodine   the iodine intake stauts of the study (sample) population
```
Comment (ernestguevarra) [inline R/detect_iodine.R:28]: No need to add a separate word (like in the @param tag) after return. You just need to describe the result that is expected.

File: R/detect_iodine.R [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+################################################################################
+#' @export
+#' @rdname detect_iodine
+#'
+
+detect_iodine_general <- function(x){
+  x <- stats::median(x, na.rm = FALSE)
+  iodine_general <- cut(x,
+                        breaks = c(-Inf, 20, 50, 100, 200, 300, Inf),
```
Comment (ernestguevarra) [inline R/detect_iodine.R:92]: Need to rethink what the lower break should be; negative values for any of the micronutrients doesn't make sense but it works. So need to figure out what a logical lower break should be

File: R/data.R (hunk 1 above)
Comment (nicholustintzaw) [reply R/data.R:70]: updated the dataset title and dataset description in the new pull request.

File: R/detect_iodine.R (hunk 2 above)
Comment (nicholustintzaw) [reply R/detect_iodine.R:28]: revised as the description of the function output.

**Pull request merged**